server/client.go

```diff
@@ -1092,53 +1097,69 @@ func (c *client) processMsg(msg []byte) {
 		if sub, ok := srv.routeSidQueueSubscriber(c.pa.sid); ok {
 			if sub != nil {
 				mh := c.msgHeader(msgh[:si], sub)
-				c.deliverMsg(sub, mh, msg)
+				if c.deliverMsg(sub, mh, msg) {
+					return
+				}
 			}
-			return
+			isRouteQsub = true
+			// for queue subscription try hard to deliver a message at least once.
```

Comment:
We want to maintain strict at-most-once semantics. I do not think this violates it, but we should update the comment IMO.

Reply:
I guess you commented from original PR, which in this PR the comments were already changed. Anyway, this comment has been removed.